nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/AbstractDatabaseFetchProcessor.java

```diff
@@ -0,0 +1,361 @@
+        dbAdapterLoader.forEach(it -> dbAdapters.put(it.getName(), it));
+
+        DB_TYPE = new PropertyDescriptor.Builder()
+                .name("db-fetch-db-type")
+                .displayName("Database Type")
+                .description("The type/flavor of database, used for generating database-specific code. In many cases the Generic type "
+                        + "should suffice, but some databases (such as Oracle) require custom SQL clauses. ")
+                .allowableValues(dbAdapters.keySet())
+                .defaultValue(dbAdapters.values().stream().findFirst().get().getName())
+                .required(true)
+                .build();
+    }
+
+    public void setup(final ProcessContext context) {
+        // Try to fill the columnTypeMap with the types of the desired max-value columns
+        final DBCPService dbcpService = context.getProperty(DBCP_SERVICE).asControllerService(DBCPService.class);
+        final String tableName = context.getProperty(TABLE_NAME).getValue();
+        final String maxValueColumnNames = context.getProperty(MAX_VALUE_COLUMN_NAMES).getValue();
+        final DatabaseAdapter dbAdapter = dbAdapters.get(context.getProperty(DB_TYPE).getValue());
+
+        try (final Connection con = dbcpService.getConnection();
+             final Statement st = con.createStatement()) {
+
+            // Try a query that returns no rows, for the purposes of getting metadata about the columns. It is possible
+            // to use DatabaseMetaData.getColumns(), but not all drivers support this, notably the schema-on-read
+            // approach as in Apache Drill
+            String query = dbAdapter.getSelectStatement(tableName, maxValueColumnNames, "1 = 0", null, null, null);
+            ResultSet resultSet = st.executeQuery(query);
+            ResultSetMetaData resultSetMetaData = resultSet.getMetaData();
+            int numCols = resultSetMetaData.getColumnCount();
+            if (numCols > 0) {
+                columnTypeMap.clear();
+                for (int i = 1; i <= numCols; i++) {
+                    String colName = resultSetMetaData.getColumnName(i).toLowerCase();
+                    int colType = resultSetMetaData.getColumnType(i);
+                    columnTypeMap.put(colName, colType);
+                }
+
+            } else {
+                throw new ProcessException("No columns found in table from those specified: " + maxValueColumnNames);
+            }
+
+        } catch (SQLException e) {
+            throw new ProcessException("Unable to communicate with database in order to determine column types", e);
+        }
+    }
+
+    protected static String getMaxValueFromRow(ResultSet resultSet,
+                                            int columnIndex,
+                                            Integer type,
+                                            String maxValueString,
+                                            String preProcessStrategy)
```

Comment:
The last parameter shouldn't be "preProcessStrategy" since that property was removed in favor of database type.